File: __tests__/fixtures/sample.ts
/**
 * A simple event system for demonstration.
 */

export interface EventHandler<T = unknown> {
  handle(event: T): void;
}

export interface Disposable {
  dispose(): void;
}

export enum LogLevel {
  Debug = 0,
  Info = 1,
  Warn = 2,
  Error = 3,
}

export type EventCallback<T> = (event: T) => void;

/** Base class for all event emitters. */
export abstract class BaseEmitter implements Disposable {
  private listeners: Map<string, Function[]> = new Map();

  abstract getName(): string;

  protected emit(event: string, data: unknown): void {
    const handlers = this.listeners.get(event);
    if (handlers) {
      handlers.forEach((fn) => fn(data));
    }
  }

  dispose(): void {
    this.listeners.clear();
  }
}

export class TypedEmitter<T> extends BaseEmitter implements EventHandler<T> {
  private level: LogLevel = LogLevel.Info;

  getName(): string {
    return "TypedEmitter";
  }

  handle(event: T): void {
    this.emit("event", event);
  }

  setLevel(level: LogLevel): void {
    this.level = level;
  }
}

export const DEFAULT_TIMEOUT = 5000;

export const createEmitter = <T>(): TypedEmitter<T> => {
  return new TypedEmitter<T>();
};

export async function waitForEvent<T>(
  emitter: BaseEmitter,
  event: string,
  timeout: number = DEFAULT_TIMEOUT,
): Promise<T> {
  return new Promise((resolve, reject) => {
    setTimeout(() => reject(new Error("timeout")), timeout);
  });
}

import { readFileSync } from "fs";

function loadConfig(path: string): Record<string, unknown> {
  const content = readFileSync(path, "utf-8");
  return JSON.parse(content);
}

File: __tests__/fixtures/sample.py
"""A simple event system for demonstration."""

import os
from pathlib import Path
from typing import Optional, List

class EventHandler:
    """Interface for handling events."""

    def handle(self, event) -> None:
        raise NotImplementedError


class BaseEmitter:
    """Base class for all event emitters."""

    name: str = "default"

    def __init__(self, max_listeners: int = 10):
        self._listeners = {}
        self._max = max_listeners

    def emit(self, event: str, data) -> None:
        handlers = self._listeners.get(event, [])
        for fn in handlers:
            fn(data)

    def dispose(self) -> None:
        self._listeners.clear()


class TypedEmitter(BaseEmitter, EventHandler):
    """Emitter that implements EventHandler."""

    def handle(self, event) -> None:
        self.emit("event", event)

    async def wait_for(self, event: str, timeout: float = 5.0) -> object:
        return await self._async_wait(event, timeout)

    @staticmethod
    def create() -> "TypedEmitter":
        return TypedEmitter()


def create_emitter(name: str = "default") -> TypedEmitter:
    emitter = TypedEmitter()
    return emitter


MAX_LISTENERS: int = 100
